src/components/audio-params.js

```diff
@@ -0,0 +1,271 @@
+      this.el.sceneEl?.systems["audio-gain"].registerSource(this);
+    }
+
+    const { enableAudioClipping, audioClippingThreshold } = window.APP.store.state.preferences;
+    this.data.clippingEnabled = enableAudioClipping !== undefined ? enableAudioClipping : CLIPPING_THRESHOLD_ENABLED;
+    this.data.clippingThreshold =
+      audioClippingThreshold !== undefined ? audioClippingThreshold : CLIPPING_THRESHOLD_DEFAULT;
+
+    if (this.data.isLocal) {
+      this.data.sourceType = SourceType.AVATAR_RIG;
+    } else if (this.el.components["media-video"]) {
+      this.data.sourceType = SourceType.MEDIA_VIDEO;
+      this.data.gain = this.el.components["media-video"].data.volume;
+      this.el.addEventListener("media-volume-changed", this.volumeUpdated.bind(this));
+    } else if (this.el.components["avatar-audio-source"]) {
+      this.data.sourceType = SourceType.AVATAR_AUDIO_SOURCE;
+      this.data.gain =
+        this.el.parentEl?.parentEl?.querySelector("[avatar-volume-controls]").components["avatar-volume-controls"]?.data
+          .volume || 1.0;
+      this.el.parentEl?.parentEl?.addEventListener("avatar-volume-changed", this.volumeUpdated.bind(this));
+    }
+  },
+
+  remove() {
+    this.normalizer = null;
+    this.el.sceneEl?.systems["audio-debug"].unregisterSource(this);
+    if (!this.data.isLocal) {
+      this.el.sceneEl?.systems["audio-gain"].unregisterSource(this);
+    }
+
+    if (this.el.components["media-video"]) {
+      this.el.removeEventListener("media-volume-changed", this.volumeUpdated);
+    } else if (this.el.components["avatar-audio-source"]) {
+      this.el.parentEl?.parentEl?.removeEventListener("avatar-volume-changed", this.volumeUpdated);
+    }
+  },
+
+  tick() {
+    const audio = this.audio();
+    if (audio) {
+      if (audio.updateMatrixWorld) {
+        audio.updateMatrixWorld(true);
+      }
+      this.data.position = new THREE.Vector3(
+        audio.panner.positionX.value,
+        audio.panner.positionY.value,
+        audio.panner.positionZ.value
+      );
+      this.data.orientation = new THREE.Vector3(
+        audio.panner.orientationX.value,
+        audio.panner.orientationY.value,
+        audio.panner.orientationZ.value
+      );
+      this.data.distanceModel = audio.panner.distanceModel;
+      this.data.rolloffFactor = audio.panner.rolloffFactor;
+      this.data.refDistance = audio.panner.refDistance;
+      this.data.maxDistance = audio.panner.maxDistance;
+      this.data.coneInnerAngle = audio.panner.coneInnerAngle;
+      this.data.coneOuterAngle = audio.panner.coneOuterAngle;
+      this.data.coneOuterGain = audio.panner.coneOuterGain;
+      this.updateDistances();
+      this.updateAttenuation();
+    }
+
+    if (this.normalizer !== null) {
+      this.normalizer.apply();
+    } else {
+      // We one only enable the Normalizer for avatar-audio-source components
+      if (this.data.sourceType === SourceType.AVATAR_AUDIO_SOURCE) {
+        this.enableNormalizer();
```

Comment:
We are calling `enableNormalizer` only in case it's not already initialized:
```
if (this.normalizer !== null) {
      this.normalizer.apply();
    } else {
      // We one only enable the Normalizer for avatar-audio-source components
      if (this.data.sourceType === SourceType.AVATAR_AUDIO_SOURCE) {
        this.enableNormalizer();
      }
    }
}
```
We can't call it on `init` as the audio element might not have been initialized yet or it might have changed and we would end up with an outdated audio element: https://github.com/mozilla/hubs/blob/565e944f5a2dac5a8b571a031a6308dabbfe370a/src/components/avatar-audio-source.js#L140